lib/core/src/msParam.cpp

```diff
@@ -43,113 +45,148 @@ addIntParamToArray( msParamArray_t *msParamArray, char *label, int inpInt ) {
  *         and inpOutBuf will be passed. If replFlag == 1, the inOutStruct
  *         and inpOutBuf will be replicated.
  */
-
-int
-addMsParamToArray( msParamArray_t *msParamArray, const char *label,
-                   const char *type, void *inOutStruct, bytesBuf_t *inpOutBuf, int replFlag ) {
-    msParam_t **newParam;
-    int len, newLen;
-
-    if ( msParamArray == NULL || label == NULL ) {
-        rodsLog( LOG_ERROR,
-                 "addMsParam: NULL msParamArray or label input" );
+int addMsParamToArray(
+    msParamArray_t* msParamArray,
+    const char* label,
+    const char* type,
+    void* inOutStruct,
+    bytesBuf_t* inpOutBuf,
+    int replFlag)
+{
+    if (!msParamArray || !label) {
+        log_msi::error("{}: received null pointer", __func__);
         return SYS_INTERNAL_NULL_INPUT_ERR;
     }
 
-    len = msParamArray->len;
+    const int len = msParamArray->len;
 
-    for ( int i = 0; i < len; i++ ) {
-        if ( msParamArray->msParam[i]->label == NULL ) {
+    // Do not allow MsParam entries sharing the same name to exist in the MsParamArray.
+    for (int i = 0; i < len; ++i) {
+        auto* msp = msParamArray->msParam[i];
+
+        // Skip the entry if the MsParam has no label (i.e. no parameter name).
+        if (!msp->label) {
             continue;
         }
-        if ( strcmp( msParamArray->msParam[i]->label, label ) == 0 ) {
-            if (!type || !msParamArray->msParam[i]->type) {
-                rodsLog(LOG_ERROR, "[%s] - type is null for [%s]", __FUNCTION__, label);
+
+        // Check the existing MsParam if its label matches the label of the new MsParam to add.
+        if (std::strcmp(msp->label, label) == 0) {
+            if (!type || !msp->type) {
+                log_msi::warn("{}: Type argument for label [{}] is a null pointer.", __func__, label);
                 continue;
             }
-            /***  Jan 28 2010 to make it not given an error ***/
-            if ( !strcmp( msParamArray->msParam[i]->type, STR_MS_T ) &&
-                    !strcmp( type, STR_MS_T ) &&
-                    !strcmp( ( char * ) inOutStruct, ( char * ) msParamArray->msParam[i]->inOutStruct ) ) {
+
+            // clang-format off
```

Comment:
It's because of the opening curly brace on L87.

I feel clang-format should never allow this:
```cpp
if (cond_1 &&
    cond_2) {
    stmt_1;
    stmt_2;
}
```
I just find that formatting to be unpleasant and clang-format doesn't provide controls for handling it.